contrib/gazebo/gazebo_sawyer_env.py

```diff
@@ -0,0 +1,499 @@
+        print("Getting robot state...")
+        self._rs = intera_interface.RobotEnable(intera_interface.CHECK_VERSION)
+        self._init_state = self._rs.state().enabled
+        print("Enabling robot...")
+        self._rs.enable()
+
+        def clean_shutdown():
+            if self._control_mode != 'position':
+                rospy.wait_for_service('/gazebo/unpause_physics')
+                try:
+                    unpause = rospy.ServiceProxy('/gazebo/unpause_physics',
+                                                 std_srvs.srv.Empty)
+                    unpause()
+                except rospy.ServiceException as e:
+                    print('/gazebo/unpause_physics service call failed')
+
+            print("Exiting...")
+            if self._rs.state().enabled:
+                print("Disabling robot...")
+                self._rs.disable()
+            # delete model
+            print("Delete gazebo models...")
+            GazeboEnv.delete_gazebo_model(model_name="cafe_table")
+            GazeboEnv.delete_gazebo_model(model_name="block")
+            GazeboEnv.delete_gazebo_model(model_name="target")
+
+            if self._control_mode != 'position':
+                rospy.wait_for_service('/gazebo/pause_physics')
+                try:
+                    pause = rospy.ServiceProxy('/gazebo/pause_physics',
+                                               std_srvs.srv.Empty)
+                    pause()
+                except rospy.ServiceException as e:
+                    print("/gazebo/pause_physics service call failed")
+
+        rospy.on_shutdown(clean_shutdown)
+
+        self.has_object = has_object
+
+        self.distance_threshold = distance_threshold
+        self.reward_type = reward_type
+        self._tip_name = end_point
+        self._limb_name = limb
+        self._limb = intera_interface.Limb(limb)
+        self._gripper = intera_interface.Gripper()
+        self._control_mode = control_mode
+        # get the joint limits
+        self._joint_limits = rospy.wait_for_message(
+            '/robot/joint_limits', JointLimits, timeout=5)
+        # set the action space
+        self._set_action_space()
+        self.target_range = target_range
+        self.target_in_the_air = target_in_the_air
+        self.obj_range = obj_range
+        self.start_angles = {
+            'right_j0': initial_qpos['right_j0'],
+            'right_j1': initial_qpos['right_j1'],
+            'right_j2': initial_qpos['right_j2'],
+            'right_j3': initial_qpos['right_j3'],
+            'right_j4': initial_qpos['right_j4'],
+            'right_j5': initial_qpos['right_j5'],
+            'right_j6': initial_qpos['right_j6']
+        }
+        self.model_states = rospy.wait_for_message(
+            "/gazebo/model_states", ModelStates, timeout=5)
+        rospy.Subscriber('/gazebo/model_states', ModelStates,
+                         self.update_model_states)
+        GazeboEnv.__init__(self, initial_qpos=initial_qpos)
```

Comment:
the ordering of a lot of this doesn't make sense. please call the superclass either directly at the top or bottom. setup related things together, and separate them logically.